proof/courier.go

```diff
@@ -543,21 +545,190 @@ type BackoffCfg struct {
 	MaxBackoff time.Duration `long:"maxbackoff" description:"The maximum backoff time to wait before retrying to deliver the proof to the receiver."`
 }
 
+// BackoffHandle is a handle to the backoff procedure.
+type BackoffHandle struct {
+	// cfg contains the backoff configuration parameters.
+	cfg *BackoffCfg
+
+	// deliveryLog is the log that the courier will use to record and query
+	// proof delivery attempts.
+	deliveryLog DeliveryLog
+}
+
+// initialDelay performs an initial delay based on the delivery log to ensure
+// that we don't spam the courier service with proof delivery attempts.
+func (b *BackoffHandle) initialDelay(ctx context.Context,
+	proofLocator Locator) error {
+
+	// Query delivery log to ensure a sensible rate of delivery attempts.
+	timestamps, err := b.deliveryLog.QueryProofDeliveryLog(
+		ctx, proofLocator,
+	)
+	if err != nil {
+		return fmt.Errorf("unable to retrieve proof delivery "+
+			"logs: %w", err)
+	}
+
+	if len(timestamps) == 0 {
+		log.Infof("No previous proof delivery attempts found "+
```

Comment:
This sounds like a `Debugf` or even `Tracef` log.

Reply:
Same for the other log outputs in this method (info should be important enough for the user to actually make it into the log).

Reply:
Agree, I think with that change details about the backoff would only get logged on failure, with the final error about # of attempts.